tests/integration/telemetry/stats/prometheus/wasm/bad_wasm_filter_test.go

```diff
@@ -0,0 +1,96 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	resource "istio.io/istio/pilot/pkg/xds/v3"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/util/retry"
+	util "istio.io/istio/tests/integration/telemetry"
+	common "istio.io/istio/tests/integration/telemetry/stats/prometheus"
+)
+
+// TestBadWasmRemoteLoad tests that bad Wasm remote load configuration won't affect service.
+// The test will set up an echo client and server, test echo ping works fine. Then apply a
+// Wasm filter which has a bad URL link, which will result as module download failure. After that,
+// verifies that echo ping could still work. The test also verifies that metrics are properly
+// recorded for module downloading failure and nack on ECDS update.
+func TestBadWasmRemoteLoad(t *testing.T) {
+	framework.NewTest(t).
+		Features("extensibility.wasm.remote-load").
+		Run(func(ctx framework.TestContext) {
+			// Test bad wasm remote load in only one cluster.
+			// There is no need to repeat the same testing logic in several different clusters.
```

Comment:
@nmittler is testing in a single cluster for a simple verification an accepted pattern in the face of MC testing?

Reply:
This exact test only cares about behavior of individual proxy re Wasm module remote load. Not sure how multi cluster is going to play a role here. Also the whole Wasm related test suite is multi-clustered, and I don't want to create a single cluster suite just for this case. so I limit it to only one cluster here. Not sure if this is an anti-pattern.. :)

Reply:
Yes. I want to know if this is OK, because there will be several telemetry integration tests with similar desires.

Reply:
Testing single cluster is fine. In general, transitioning our prow jobs over to multicluster has been about simply reducing the number of prow jobs. If you had a single prow job dedicated to single cluster testing, that should be fine.